### DISCUSSION TITLE
Adding EMG sensor ESP32 controller to control the CrazyFlie drone

### DISCUSSION THREAD

**Discussion by qurrpurr** [bitcraze/discussions#1562] [Q&A] [answered]
Hello. Per title, I am trying to work on developing EMG controller to substitute the drone controller in Crazyflie. I have done the simple coding of the EMG controller when working with a self made Arduino ESP32 drone, but this time I am trying to make it work on the Espressif ESP-drone Crazyflie drone using this opensource code
 [https://github.com/espressif/esp-drone](url)
 which have more complicated coding than what I am used to (I am still quite a beginner in C coding and only used to Arduino coding). 

My questions are
1) is it possible? If yes, which file do I need to look and add the coding? 
2) Any examples/tutorials of c code work to design the controller?

If you need any more details please let me know! I will be grateful for any help 

I also have done some research and found out this discussion seems to be similar with [(https://forum.bitcraze.io/viewtopic.php?t=5311](url), but if there are still some suggestions please feel free to write down comments

**Comment by gemenerik** [comment, marked as answer]
Thank you for reaching out. From your description, it seems that you are using hardware that is not ours, and the firmware running on it, while based on a ported version of our open-source software, is not directly our firmware. Unfortunately, we are unable to provide support for hardware or firmware that is not part of our official products. I recommend contacting the developers or vendors responsible for the hardware and firmware you're using for further assistance.
